packages/yoroi-extension/app/utils/formatters.js

```diff
@@ -1,15 +1,38 @@
 // @flow
 import BigNumber from 'bignumber.js';
 
+export const removeZeros: (string => string) = (amount: string): string => {
+  let sliceZerosIdx = amount.length
+  const amountLength = amount.length
+  for (let i = 0; i < amountLength; i++) {
+    if (Number(amount[i]) === 0 && sliceZerosIdx === amountLength) sliceZerosIdx = i 
+    if (Number(amount[i]) !== 0) sliceZerosIdx = amountLength
+  }
+  return amount.slice(0, sliceZerosIdx)
+}
+
 export function splitAmount(
   amount: BigNumber,
   decimalPlaces: number,
 ): [string, string] {
   const valString = amount.toFormat(decimalPlaces);
   const startIndex = valString.length - decimalPlaces;
-  return [valString.substring(0, startIndex), valString.substring(startIndex)];
+  let beforeDecimal = valString.substring(0, startIndex)
+  const afterDecimal = removeZeros(valString.substring(startIndex))
+  // Remove the dots if no decimals
+  if (Number(afterDecimal) === 0) {
+    beforeDecimal = beforeDecimal.slice(0, beforeDecimal.length - 1)
+  }
+  return [beforeDecimal, afterDecimal]
```

Comment:
Note when `amount` is `new BigNumber('1')` and `decimalPlaces` is `0`, this function outputs `['', '']`, but it should output `['1', '']`. Please fix and add a test file `formatters.test.js` to test it.